src/modules/jcms_article_fragment/src/ArticleFragmentAccessControlHandler.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Drupal\jcms_article_fragment;
+
+use Drupal\Core\Entity\EntityAccessControlHandler;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Access\AccessResult;
+
+/**
+ * Access controller for the Article fragment entity.
+ *
+ * @see \Drupal\jcms_article_fragment\Entity\ArticleFragment.
+ */
+class ArticleFragmentAccessControlHandler extends EntityAccessControlHandler {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function checkAccess(EntityInterface $entity, $operation, AccountInterface $account) {
+    /** @var \Drupal\jcms_article_fragment\Entity\ArticleFragmentInterface $entity */
+    switch ($operation) {
+      case 'view':
+        if (!$entity->isPublished()) {
+          return AccessResult::allowedIfHasPermission($account, 'view unpublished article fragment entities');
+        }
+        return AccessResult::allowedIfHasPermission($account, 'view published article fragment entities');
+
+      case 'update':
+        return AccessResult::allowedIfHasPermission($account, 'edit article fragment entities');
+
+      case 'delete':
```

Comment:
I've commented out the code that allows deletion so that we can reimplement it later.